### PR TITLE
Weapons

### PR DESCRIPTION
**Pull request by Prakkkmak**

### CODE REVIEW

**Comment by Prakkkmak** [comment]
[Weapons](https://trello.com/c/ofDkvO2s/38-weapons)

File: code/Items/ItemStackWeapon.cs [hunk 1]
```diff
@@ -1,15 +1,136 @@
...
+        }
+
+        protected virtual void Reload()
+        {
+        }
+
+        public virtual bool CanPrimaryAttack()
+        {
+
+            if (!Owner.IsValid() || !Input.Down(InputButton.Attack1)) return false;
```
Comment (Nowp) [inline code/Items/ItemStackWeapon.cs:56]:
```suggestion
        {
            if (!Owner.IsValid() || !Input.Down(InputButton.Attack1)) return false;
```

File: code/Items/ItemStackWeapon.cs [hunk 2]
```diff
@@ -1,15 +1,136 @@
...
+            return Owner.IsValid() && Input.Down(InputButton.Reload);
+        }
+
+        protected virtual void Reload()
+        {
+        }
+
+        public virtual bool CanPrimaryAttack()
+        {
+
```
Comment (Nowp) [inline code/Items/ItemStackWeapon.cs:55]: How does it check that the weapon is selected ? Is it Owner.IsValid ?

Comment (Prakkkmak) [reply code/Items/ItemStackWeapon.cs]: owner.valid check if the owner exists so yeah

**Review by Nowp — APPROVED** [review]
It will be overwritten by my item rework anyway

**Pull request merged**